cmd/datum-authorization-webhook/app/serve.go

```diff
@@ -0,0 +1,157 @@
+				))
+				otel.SetTextMapPropagator(propagation.NewCompositeTextMapPropagator(
+					propagation.TraceContext{},
+					propagation.Baggage{},
+				))
+			}
+
+			dialOptions := []grpc.DialOption{
+				grpc.WithStatsHandler(otelgrpc.NewClientHandler()),
+				grpc.WithChainUnaryInterceptor(
+					func(ctx context.Context, method string, req, reply any, cc *grpc.ClientConn, invoker grpc.UnaryInvoker, opts ...grpc.CallOption) error {
+						logger := log.Log.WithName("grpc_client")
+						logger.Info(method, ".request: ", protojson.Format(req.(proto.Message)))
+						err := invoker(ctx, method, req, reply, cc, opts...)
+						logger.Info(method, ".response: ", protojson.Format(reply.(proto.Message)))
+						return err
+					},
+				),
+			}
+
+			if iamInsecure {
+				dialOptions = append(dialOptions, grpc.WithTransportCredentials(insecure.NewCredentials()))
+			} else {
+				dialOptions = append(dialOptions, grpc.WithTransportCredentials(credentials.NewTLS(nil)))
+			}
+
+			iamConnection, err := grpc.NewClient(iamEndpoint, dialOptions...)
+			if err != nil {
+				return fmt.Errorf("failed to create new IAM client: %w", err)
+			}
+			defer iamConnection.Close()
+
+			log.SetLogger(zap.New(zap.JSONEncoder()))
+
+			iamClient := iamv1alphagrpc.NewAccessCheckClient(iamConnection)
+
+			entryLog := log.Log.WithName("entrypoint")
+
+			restConfig, err := config.GetConfig()
+			if err != nil {
+				return fmt.Errorf("failed to get rest config: %s", err)
+			}
+
+			runtimeScheme := runtime.NewScheme()
+			v1beta1.AddToScheme(runtimeScheme)
+
+			// Setup a Manager
+			entryLog.Info("setting up manager")
+			mgr, err := manager.New(restConfig, manager.Options{
+				Scheme: runtimeScheme,
+				Metrics: server.Options{
+					BindAddress: ":8999",
+				},
+				WebhookServer: webhook.NewServer(webhook.Options{
+					CertDir:  certDir,
+					CertName: certFile,
+					KeyName:  keyFile,
+				}),
+			})
+			if err != nil {
+				return fmt.Errorf("failed to setup manager: %s", err)
+			}
+
+			// Setup webhooks
+			entryLog.Info("setting up webhook server")
+			hookServer := mgr.GetWebhookServer()
+
+			entryLog.Info("registering webhooks to the webhook server")
+
+			hookServer.Register("/project/v1alpha/{project...}", authwebhook.NewAuthorizerWebhook(&iam.ProjectControlPlaneAuthorizer{
+				IAMClient: iamClient,
+			}))
+			hookServer.Register("/core/v1alpha/webhook", authwebhook.NewAuthorizerWebhook(&iam.CoreControlPlaneAuthorizer{
+				IAMClient: iamClient,
+			}))
+
+			return mgr.Start(context.Background())
+		},
+	}
+
+	serveCmd.Flags().StringVar(&iamEndpoint, "iam-endpoint", "", "Endpoint to use for connecting to the datum gRPC API endpoint")
+	serveCmd.Flags().BoolVar(&iamInsecure, "iam-endpoint-insecure", false, "Whether the use an insecure connection when export")
+
+	serveCmd.Flags().StringVar(&tracingEndpoint, "tracing-endpoint", "", "Endpoint to send OpenTelemetry traces too. Tracing is disabled if not provided.")
```

Comment:
I think the `OTEL_EXPORTER_OTLP_ENDPOINT` environment variable would work for this as well. See https://opentelemetry.io/docs/languages/sdk-configuration/otlp-exporter/

Reply:
Good call, removed those options.